MC/run/PWGGAJE/run_dirgamma.sh

```diff
@@ -1,39 +1,86 @@
 #!/usr/bin/env bash
 
 # Generate gamma-jet events, Pythia8 in a given pt hard bin.
-# run_dirgamma.sh n_pthatbin
+# Execute: ./run_dirgamma.sh 
+# Set at least before running PTHATBIN with 1 to 6
 
-set -x
 
-SIGEVENTS=10
-NWORKERS=2
-MODULES=
-RNDSEED=0    # [default = 0] time-based random seed
+#set -x 
 
-# generate Pythia8 gamma-jet configuration
+# ----------- LOAD UTILITY FUNCTIONS --------------------------
+. ${O2_ROOT}/share/scripts/jobutils.sh
+
+# ----------- START ACTUAL JOB  ----------------------------- 
+
+RNDSEED=${RNDSEED:-0}   # [default = 0] time-based random seed
+
+NSIGEVENTS=${NSIGEVENTS:-20}
+NTIMEFRAMES=${NTIMEFRAMES:-5}
+NWORKERS=${NWORKERS:-8}
+MODULES="--skipModules ZDC" #"PIPE ITS TPC EMCAL"
+CONFIG_ENERGY=${CONFIG_ENERGY:-5020.0}
+CONFIG_NUCLEUSA=${CONFIG_NUCLEUSA:-2212}
+CONFIG_NUCLEUSB=${CONFIG_NUCLEUSB:-2212}
 
 # Define the pt hat bin arrays
 pthatbin_loweredges=(5 11 21 36 57 84)
 pthatbin_higheredges=(11 21 36 57 84 -1)
 
-# Define environmental vars for pt binning
-PTHATBIN=$1 #set it here or externally? Add protection out of array?
+# Recover environmental vars for pt binning
+#PTHATBIN=${PTHATBIN:-1} 
+
+if [ -z "$PTHATBIN" ]; then
+    echo "Pt-hat bin (env. var. PTHATBIN) not set, abort."
+    exit 1
+fi
 
 PTHATMIN=${pthatbin_loweredges[$PTHATBIN]}
 PTHATMAX=${pthatbin_higheredges[$PTHATBIN]}
 
+# We will collect output files of the workflow in a dedicated output dir
+# (these are typically the files that should be left-over from a GRID job)
+[ ! -d output ] && mkdir output
+
+copypersistentsimfiles() {
+  simprefix=$1
+  outputdir=$2
+  cp ${simprefix}_Kine.root ${simprefix}_grp.root ${simprefix}*.ini ${outputdir}
+}
+
+# Loop over timeframes
+for tf in `seq 1 ${NTIMEFRAMES}`; do
```

Comment:
Hi @gconesab : I think I had in mind a much more minor change: just prepend the elemental stages (o2-sim) by the "taskwrapper" keyword. My intention was that you get familiar with it's features and provide possible feedback.

You went further and made this into full embedding examples. Indeed, it in the end this repetitive code should be handled by a mother sim workflow.

I have no hard feelings about this. We can do the changes for now or you may keep it more simple. What do you think?

Reply:
OK, then let's keep it simple, also the timeframe loop should go out right?

Reply:
yep